### PR TITLE
PM-2153 Update User Confirm Dialog

### PR DESCRIPTION
**Pull request by KiruthigaManivannan**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [x] Other
```

## Objective

Migrate the User Confirm Dialog component to use the component library.

## Code changes
- **user-confirm.component.html:** Converted Modal to use its Equivalent Dialog from library
- **user-confirm.component.ts:** Added dialog open method , its params and dialog result
- **base.people.component.ts:** Added dialog open logic, captured dialog result 

## Screenshots

https://github.com/bitwarden/clients/assets/162679756/02faa5b7-7241-4d0b-9b6d-2d0cad47badd

### CODE REVIEW

File: apps/web/src/app/admin-console/common/base.people.component.ts [hunk 1]
```diff
@@ -356,25 +359,22 @@ export abstract class BasePeopleComponent<
...
+            name: this.userNamePipe.transform(user),
+            userId: user != null ? user.userId : null,
+            publicKey: publicKey,
           },
-        );
+        });
+        const result = await lastValueFrom(dialogRef.closed);
+        if (result === UserConfirmDialogResult.Confirmed) {
+          try {
+            const response = confirmUser(publicKey);
```
Comment (r-tome) [inline apps/web/src/app/admin-console/common/base.people.component.ts:372]: The `confirmUser` logic should be placed inside the `Submit` action on the `UserConfirmComponent` so the dialog only closes after the action is performed.

**Review by r-tome — COMMENTED** [review]
Everything looks good to me except the dialog submit logic. It could be out of scope because I think that is how its currently implemented. What are your thoughts @willmartian?

File: apps/web/src/app/admin-console/common/base.people.component.ts (hunk 1 above)
Comment (willmartian) [reply apps/web/src/app/admin-console/common/base.people.component.ts:372]: Yes agreed. That more closely matches what existed previously: the promise happens when the dialog is open, and the dialog only closes if the promise successfully resolves.

Comment (KiruthigaManivannan) [reply apps/web/src/app/admin-console/common/base.people.component.ts:372]: @r-tome  @willmartian  But confirmUser method is in base people component. So how can we call it inside UserConfirmComponent

Comment (KiruthigaManivannan) [reply apps/web/src/app/admin-console/common/base.people.component.ts:409]: @r-tome  @willmartian  Can you please advise on the above comment

Comment (r-tome) [reply apps/web/src/app/admin-console/common/base.people.component.ts:409]: @willmartian exactly what I was thinking!

**Review by r-tome — APPROVED** [review]

**Pull request merged**